File: 4/problem_four.rb
# A palindromic number reads the same both ways. The largest palindrome made
# from the product of two 2-digit numbers is 9009 = 91 × 99.
#
# Find the largest palindrome made from the product of two 3-digit numbers.


def largest_palindrome(n)
  start = ('1' + '0' * (n - 1)).to_i
  _end = (start.to_s + '0').to_i
  (start.._end).reduce(0) do |largest_pal, num|
    # we're always going to multiply 10 x 10, 11 x 11
  end
end

def palindrome?(n)
  n.to_s.reverse == n.to_s
end

# def assign_largest_pal(max, current_result)
#   if current_result > max
#     current_result
#   else
#     largest_pal
#   end
# end

puts largest_palindrome(2)

File: 4/problem_four.js
// A palindromic number reads the same both ways. The largest palindrome made
// from the product of two 2-digit numbers is 9009 = 91 × 99.
//
// Find the largest palindrome made from the product of two 3-digit numbers.

function largestPalindrome() {
  var largestPal = 0;

  for (var i = 100; i < 1000; i++) {
    for (var j = 100; j < 1000; j++) {
      var result = i * j;
      if (isPalindrome(result) && result > largestPal)
        largestPal = result;
    }
  }

  return largestPal;
}

function isPalindrome(n) {
  return n.toString() === n.toString().split('').reverse().join('');
}

console.log(largestPalindrome());
